cashier/src/main/java/com/getkeepsafe/cashier/Cashier.java

```diff
@@ -300,7 +314,9 @@ public void dispose() {
    * the {@link Vendor}
    */
   public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
-    return vendor.onActivityResult(requestCode, resultCode, data);
+    boolean purchaseResultReceived = vendor.onActivityResult(requestCode, resultCode, data);
+    sPurchaseInProgress = !purchaseResultReceived;
```

Comment:
What happens then `onActivityResult` returns `false`? Is the fact that the purchase activity returns a result not signal enough that a new purchase should be possible?